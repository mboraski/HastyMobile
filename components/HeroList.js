// Third Party Imports
import React, { Component } from 'react';
import { ScrollView, StyleSheet } from 'react-native';
import _ from 'lodash';

// Relative Imports
import HeroDetail from './HeroDetail';
import { emY } from '../utils/em';

class HeroList extends Component {
    state = {
        heroes: []
    };

    componentWillMount() {
        this.setState({
            heroes: Array(4).fill().map((e, i) => ({
                id: i,
                name: 'Jessica Morgan',
                type: i % 2 ? 'Instant' : 'Driver',
                delivery_time: 4,
                image: 'https://facebook.github.io/react/img/logo_og.png'
            }))
        });
    }

    renderHeroes() {
        return _.map(this.state.heroes, hero =>
            <HeroDetail
                key={hero.id}
                hero={hero}
            />
        );
    }

    render() {
        return (
            <ScrollView contentContainerStyle={styles.container}>
                {this.renderHeroes()}
            </ScrollView>
        );
    }
}

const styles = StyleSheet.create({
    container: {
    }
});

export default HeroList;
