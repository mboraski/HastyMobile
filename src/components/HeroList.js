// Third Party Imports
import React, { Component } from 'react';
import { View, StyleSheet } from 'react-native';
import { connect } from 'react-redux';

// Relative Imports
import HeroDetail from './HeroDetail';

class HeroList extends Component {
    // heroes: [
    //     {
    //         id: 1,
    //         name: 'Jessica Morgan',
    //         type: 'Instant',
    //         delivery_time: 4,
    //         image: 'https://facebook.github.io/react/img/logo_og.png'
    //     },
    //     {
    //         id: 2,
    //         name: 'Matt Littlefield',
    //         type: 'Fast',
    //         delivery_time: 16,
    //         image: 'https://facebook.github.io/react/img/logo_og.png'
    //     }
    // ],

    render() {
        const { hero } = this.props;
        let result;
        console.log('hero: ', hero);
        if (hero) {
            result = (
                <View style={styles.container}>
                    <HeroDetail
                        hero={hero}
                    />
                </View>
            );
        } else {
            result = null;
        }
        return result;
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff'
    }
});

const mapStateToProps = state => ({
    hero: state.order.hero
});

const mapDispatchToProps = {};

export default connect(mapStateToProps, mapDispatchToProps)(HeroList);
