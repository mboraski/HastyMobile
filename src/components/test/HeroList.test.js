import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';

import { HeroList } from '../HeroList';

describe('HeroList', () => {
    it('renders correctly', () => {
        const render = shallow(<HeroList />);
        expect(render).toMatchSnapshot();
    });
});
