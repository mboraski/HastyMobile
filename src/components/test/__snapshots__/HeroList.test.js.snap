// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroList renders correctly 1`] = `
<ScrollViewMock
  contentContainerStyle={Object {}}
/>
`;
